ninja-core/src/main/java/ninja/Router.java

```diff
@@ -43,6 +43,9 @@
      * @param clazz The controllerClass e.g. ApplicationController.class
      * @param methodName the methodName of the class e.g. "index"
      * @return The final url (without server, and without any prefixes)
+     * @deprecated Reverse routing in the Router is not validated and does not
+     *      URL-escape path or query parameters. Use <code>ninja.ReverseRouter</code>
+     *      to build your reverse routes.
      */
```

Comment:
Not sure how i forgot the annotation.  Fixed.  We have a LOT of these in our code and it would be difficult to upgrade everything -- backwards compat here is important.  We can remove later in Ninja 7.